exercises/diffie-hellman/diffie_hellman_test.exs

```diff
@@ -0,0 +1,98 @@
+if !System.get_env("EXERCISM_TEST_EXAMPLES") do
+  Code.load_file("diffie_hellman.exs", __DIR__)
+end
+
+ExUnit.start
+ExUnit.configure exclude: :pending, trace: true
+
+defmodule DiffieHellmanTest do
+  use ExUnit.Case
+
+  #@tag :pending
+  test "private key should be random between 1 and P-1, inclusive" do
+    prime_p = 23
+
+    generated_private_keys = 1..100 |> Enum.map(fn _i -> DiffieHellman.generate_private_key(prime_p) end)
```

Comment:
This was basically just to call the function 100 times and ensure some randomness to it. Didn't need to be in two tests, so I rewrote it a bit after changing the tests to:

- Does it generate a number between 1-(P-1)? (Should handle your previous request)
- Does it handle arbitrarily large P?
- Is the number random?

Now that there's only one test using it, I cleared up what it's doing by not doing an `Enum.map` over an unused number, but calling `Stream.repeatedly` and taking 100 digits. I think it's clearer, and can move it to a helper if you still want, but don't think it belongs in the DiffieHellman class itself.

I also moved the hints to `HINTS.md`, so that should cover your concerns, I hope.